inference-engine/src/auto_plugin/auto_plugin.cpp

```diff
@@ -0,0 +1,206 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <string>
+#include <vector>
+#include <memory>
+#include <map>
+#include <unordered_set>
+
+#include <ie_metric_helpers.hpp>
+#include <ie_core.hpp>
+#include <threading/ie_executor_manager.hpp>
+#include <ie_algorithm.hpp>
+
+#include "auto_plugin.hpp"
+
+namespace AutoPlugin {
+namespace {
+    ConfigType mergeConfigs(ConfigType config, const ConfigType& local) {
+        for (auto && kvp : local) {
+            config[kvp.first] = kvp.second;
+        }
+        return config;
+    }
+
+    DeviceInformation SelectDevice(const std::vector<DeviceInformation>& metaDevices) {
+        for (auto& item : metaDevices) {
+            if (item.deviceName.find("CPU") == 0) {
```

Comment:
what if CPU device is disabled? E.g. not compiled / not installed. We need a way to select any other devic

Reply:
yes, for this first PR, we will only select CPU, about the limitation feature and select logic or error handling, we would pull another PR asap.

Reply:
MR for updating device selection logic here: [Optimize the device selection logic for auto-plugin](https://github.com/openvinotoolkit/openvino/pull/5549/commits);

Will loop until find the device can be loaded:

`while (!metaDevices.empty()) {
        selectedDevice = SelectDevice(network, metaDevices, optCap);
        try {
            // LoadNetwork on selectedDevice ...
            break;
        } catch(// Cannot Load one selectedDevice...) {
          // Erase this device from metaDevices
        }
    }`